### PR TITLE
support ErrorBoundary in asBaseComponent

### PR DESCRIPTION
**Pull request by lidord-wix**
## Description
Support ErrorBoundary in asBaseComponent

## Changelog
Support ErrorBoundary in asBaseComponent

### CODE REVIEW

File: src/commons/asBaseComponent.tsx [hunk 1]
```diff
@@ -25,17 +25,30 @@ function asBaseComponent<PROPS, STATICS = {}>(WrappedComponent: React.ComponentT
...
+    state = {
+      error: false
+    };
+
     static getThemeProps = (props: any, context: any) => {
       return Modifiers.getThemeProps.call(WrappedComponent, props, context);
     }
 
+    static getDerivedStateFromError(error: any) {
+      UIComponent.defaultProps.onError(error, WrappedComponent.defaultProps);
```
Comment (ethanshar) [inline src/commons/asBaseComponent.tsx:37]: Did you test what happens in public uilib when there's an error? (:

File: src/commons/asBaseComponent.tsx [hunk 2]
```diff
@@ -25,17 +25,30 @@ function asBaseComponent<PROPS, STATICS = {}>(WrappedComponent: React.ComponentT
...
+
     render() {
       const themeProps = BaseComponent.getThemeProps(this.props, this.context);
       const modifiers = Modifiers.generateModifiersStyle(undefined, themeProps);
       // TODO: omit original modifiers props (left, right, flex, etc..)
       // Because they throws an error when being passed to RNView on Android
       const {forwardedRef, ...others} = themeProps;
-      return <WrappedComponent {...others} modifiers={modifiers} ref={forwardedRef}/>;
+      return this.state.error ? (
+        UIComponent.defaultProps.renderError
```
Comment (ethanshar) [inline src/commons/asBaseComponent.tsx:48]: Same here, better check how it works in public..

**Comment by ethanshar** [comment]
Looks great !
See my comments

File: src/commons/asBaseComponent.tsx (hunk 1 above)
Comment (lidord-wix) [reply src/commons/asBaseComponent.tsx]: Handled now :)

File: src/commons/asBaseComponent.tsx (hunk 2 above)
Comment (lidord-wix) [reply src/commons/asBaseComponent.tsx]: Fixed :)

**Review by ethanshar — APPROVED** [review]

**Pull request merged**